### ISSUE TITLE
How to run guvicorn as a service?

### ISSUE THREAD

**Issue by ligantx** [tiangolo/fastapi#5387]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
gunicorn main:app --workers 4 --worker-class uvicorn.workers.UvicornWorker --bind 0.0.0.0:80
```

### Description

Although the documentation (which btw is excellent) refers to deployment with Docker, and after online search, i cannot find how i will run guvicorn command:
`gunicorn main:app --workers 4 --worker-class uvicorn.workers.UvicornWorker --bind 0.0.0.0:80` as a service (without Docker). 
Is there a recommended way for production? I found that in linux i can create a systemd service and run it always and after a crash. How i can to the same in windows?
And if i do that, do i need a reverse proxy (i like caddy) to run on top?

### Operating System

Linux, Windows, macOS, Other

### Operating System Details

raspberry pi os

### FastAPI Version

0.79.0

### Python Version

Python 3.9.12

**Comment by iudeen** [comment]
You can't run Gunicorn in windows.

**Comment by ligantx** [comment]
> found that in linux i can create a systemd service and run it always and after a crash. How i can to the same in windows?

I found that windows runs gunicorn only under Docker. 

But in linux i should create my own systemd service and then run reverse proxy like caddy on top?

**Comment by csrgxtu** [comment]
> Is there a recommended way for production? I found that in linux i can create a systemd service and run it always and after a crash. How i can to the same in windows?

i use Linux or Mac as dev enviroment. but if you searching for auto restart after exception, you should consider `supervisor`, which is a python process monitor tool which guarateens your app wont stop.

**Comment by ligantx** [comment]
I would prefer systemd as there are no dependencies and they exist by default in linux. There is no way to make this work?

**Comment by jgould22** [comment]
> there are no dependencies and they exist by default in linux

This is unlikely to be a true statement, regardless you can do as you described and create a systemd service to run that command.  Whether you need to run a reverse proxy in front depends entirely on your application.

**Comment by jkazimierczak** [comment]
> i finally manage to make it with systemd for linux

If you managed to make a systemd file for your service then describe how you've done it. If someone stumbles upon this issue they will also know how to solve it.

**Comment by ligantx** [comment]
i did something similar to [this](http://www.learningaboutelectronics.com/Articles/How-to-create-a-gunicorn-service-file-in-linux.php)

**Comment by jkazimierczak** [comment]
I think it's good to post the solution in the issue. Linking to external sites poses the risk of site going down and making the solution disappear. After you post the solution you can also close the issue.

**Comment by 4mishra** [comment]
Another possibility I see, using msys2 or cygwin on windows, Grab the gunicorn source source code and compile it with simple command:
python setup.py install

and then run gunicorn application happily on windows

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

If that solves the original problem, then you can close this issue @ligantx  ✔️

**Comment by ligantx** [comment]
> I think it's good to post the solution in the issue. Linking to external sites poses the risk of site going down and making the solution disappear. After you post the solution you can also close the issue.

You are right. So, [here](https://github.com/ligantx/systemd-gunicorn-uvicorn/blob/main/README.MD) is my approach on how i created the systemd (spending a day of reading documentations etc. :P) with some additional useful info. I wish it will help other newcomers.

So, i wrote it the best i could. Please notify if you find any error, as i translated the instructions from my native language.
